src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -4111,10 +4111,8 @@ internal static IEnumerable<CompletionResult> CompleteFilename(CompletionContext
                 var relativePaths = context.GetOption("RelativePaths", @default: defaultRelative);
                 var useLiteralPath = context.GetOption("LiteralPaths", @default: false);
 
-                if (useLiteralPath && LocationGlobber.StringContainsGlobCharacters(wordToComplete))
-                {
-                    wordToComplete = WildcardPattern.Escape(wordToComplete, Utils.Separators.StarOrQuestion);
-                }
+                // We should always escape '[' and ']' if present.
+                wordToComplete = WildcardPattern.Escape(wordToComplete, Utils.Separators.StarOrQuestion);
```

Comment:
It seems like the code was correct before - only when we know we have -LiteralPath.